include/zephyr/drivers/charger.h

```diff
@@ -0,0 +1,190 @@
+/*
+ * Copyright 2023 Cirrus Logic, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_DRIVERS_CHARGER_H_
+#define ZEPHYR_INCLUDE_DRIVERS_CHARGER_H_
+
+/**
+ * @brief Charger Interface
+ * @defgroup charger_interface Charger Interface
+ * @ingroup io_interfaces
+ * @{
+ */
+
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+
+#include <zephyr/device.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif /* __cplusplus */
+
+/**
+ * @brief Runtime Dynamic Battery Parameters
+ */
+enum charger_props {
+	/** Indicates if external supply is present for the charger */
+	CHARGER_PROP_ONLINE = 0,
+	/** Reports whether or not a battery is present */
+	CHARGER_PROP_PRESENT,
+	/** Represents the charging status of the charger */
+	CHARGER_PROP_STATUS,
+	/** Reserved to demark end of common charger properties */
+	CHARGER_PROP_COMMON_COUNT,
+	/**
+	 * Reserved to demark downstream custom properties - use this value as the actual value may
+	 * change over future versions of this API
+	 */
+	CHARGER_PROP_CUSTOM_BEGIN = CHARGER_PROP_COMMON_COUNT + 1,
+	/** Reserved to demark end of valid enum properties */
+	CHARGER_PROP_MAX = UINT16_MAX,
+};
```

Comment:
Each property should document the expected returned type, otherwise the API is fragile IMO.
An kind of "variant" type could be useful here, and possibly be a generic Zephyr lib on its own, used in other places.

Reply:
Agreed. This seems essential for understanding how to use the API.

Reply:
ack